### PR TITLE
Adds 'size' option to 'columns' input of DataTable

### PR DESCRIPTION
**Pull request by raj-chelur**
#### What does this PR do?
Adds 'size' option to 'columns' input of DataTable.

#### What testing has been done on this PR?
Added new storybook stories and updated jest tests.

#### How should this be manually tested?
See storybook - DataTable > ColumnSizes

#### Any background context you want to provide?
The Table components in grommet allows specifying a size for the columns/cells, but this was missing in DataTable. We need control on certain column sizes to avoid wrap or truncate of text.

#### Screenshots (if appropriate)
image

image

#### Do the grommet docs need to be updated?
Done.

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Backward compatible - size is an optional input in 'columns' spec for DataTable

### CODE REVIEW

**Comment by ericsoderberghp** [comment]
This is looking good. Have you checked how the combination of `size` in `columns` interacts with `resizeable`?

**Comment by raj-chelur** [comment]
> This is looking good. Have you checked how the combination of `size` in `columns` interacts with `resizeable`?

Thanks! A change was required to pass size to Header as well. I have added a new story to test size with resizable. 'size' if specified is acting as the initial width of the column. I will update the pull request:

Default widths on first load:

image

After resizing a few columns:

image

File: src/js/themes/__tests__/__snapshots__/theme-test.js.snap [hunk 1]
```diff
@@ -1567,7 +1567,7 @@ exports[`Grommet hpe theme 1`] = `
   outline: none;
   max-width: 100%;
   background: #01A982;
-  color: #FFFFFF;
+  color: #EEEEEE;
```
Comment (ShimiSun) [inline src/js/themes/__tests__/__snapshots__/theme-test.js.snap]: This `color` shouldn't have changed on the theme test, will you be willing to explore why this change occur?

**Comment by raj-chelur** [comment]
> Thanks for the PR!
> Can you also add a test on DataTable test file for this new `size` prop?
> We are trying to cover all new props with a snapshot or test.

Hi,

I had included 3 tests in DataTable-test.js. Let me know if there are other tests that need to be updated ...

https://github.com/grommet/grommet/pull/3901/files?file-filters%5B%5D=.js&file-filters%5B%5D=.snap&file-filters%5B%5D=.ts#diff-393e116acabc81a64f0225f07770e0da

File: src/js/themes/__tests__/__snapshots__/theme-test.js.snap (hunk 1 above)
Comment (raj-chelur) [reply src/js/themes/__tests__/__snapshots__/theme-test.js.snap]: Not sure what happened there ... may be I mixed up npm and then yarn in my workflow. We are using 'npm' in other projects, so its quite possible I mixed it up.

I have however reverted these changes and its not included in the final commit.

Comment (ShimiSun) [reply src/js/themes/__tests__/__snapshots__/theme-test.js.snap]: Thanks!

File: src/js/components/DataTable/stories/ColumnSize.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  {
+    name: 'Tracy Kimbery Mccrary Jona Kinsey',
+    location: 'San Francisco',
+    date: '2018-06-10',
+    percent: 10,
+    paid: 2345,
+  },
+];
+
+const columnsThemeSize = [
```
Comment (ShimiSun) [inline src/js/components/DataTable/stories/ColumnSize.js:66]: Not sure what the user should understand from the name that includes theme while there is no theme involved in this example. Could you please explain the thought behind it?

File: src/js/components/DataTable/stories/ColumnSize.js [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    </Box>
+  </Grommet>
+);
+
+const ExampleResizable = () => (
+  <Grommet theme={grommet}>
+    <Box align="center" pad="medium">
+      <Heading level="3">Large table with resizeable</Heading>
+      <DataTable
+        columns={columnsThemeSize}
```
Comment (ShimiSun) [inline src/js/components/DataTable/stories/ColumnSize.js:172]: I was trying to resize between the first and second column and it wasn't working. is it a bug?

File: src/js/components/DataTable/stories/ColumnSize.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        data={DATA}
+        step={10}
```
Comment (ShimiSun) [inline src/js/components/DataTable/stories/ColumnSize.js:174]: I don't think we are using `step` in those examples, let's remove it.

File: src/js/components/DataTable/stories/ColumnSize.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        resizeable
+      />
+    </Box>
+
+    <Box align="center" pad="medium">
+      <Heading level="3">Small table with resizeable</Heading>
+      <DataTable
+        columns={columnsReize}
+        data={DATA}
+        step={10}
```
Comment (ShimiSun) [inline src/js/components/DataTable/stories/ColumnSize.js:185]: I don't think we are using `step` in those examples, let's remove it.

File: src/js/components/DataTable/stories/ColumnSize.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+        primaryKey={false}
+        resizeable
+      />
+    </Box>
+  </Grommet>
+);
+storiesOf('DataTable', module).add('Column Sizes', () => <Example />);
+
+storiesOf('DataTable', module).add('Column Sizes resizeable', () => (
```
Comment (ShimiSun) [inline src/js/components/DataTable/stories/ColumnSize.js:194]: From user feedback, we found that separating the files of the stories is more readable and easier to navigate inside the code. 
Could you please separate this file into two files for each story? you can share the common data with importing it from one file to another and add a comment with link of where this data is taken from. It will help the users navigate.

**Comment by ShimiSun** [comment]
@raj-chelur any updates on this PR? would you need any further direction on this one?

File: src/js/components/DataTable/stories/ColumnSize.js (hunk 2 above)
Comment (raj-chelur) [reply src/js/components/DataTable/stories/ColumnSize.js:66]: I meant the sizes "small", "medium", "large" etc the come from the theme definition.

Please suggest if there's a better name for these size definitions - like say "standard sizes"?

File: src/js/components/DataTable/stories/ColumnSize.js (hunk 3 above)
Comment (raj-chelur) [reply src/js/components/DataTable/stories/ColumnSize.js:172]: Good catch! I have now made widths set by resizing override the column size specified.

Checked that it now works same as the other resizable story (Tunable)

File: src/js/components/DataTable/stories/ColumnSize.js (hunk 4 above)
Comment (raj-chelur) [reply src/js/components/DataTable/stories/ColumnSize.js:174]: Done!

File: src/js/components/DataTable/stories/ColumnSize.js (hunk 5 above)
Comment (raj-chelur) [reply src/js/components/DataTable/stories/ColumnSize.js:185]: Done!

File: src/js/components/DataTable/stories/ColumnSize.js (hunk 6 above)
Comment (raj-chelur) [reply src/js/components/DataTable/stories/ColumnSize.js:194]: Done!

File: src/js/components/DataTable/stories/ColumnSize.js (hunk 2 above)
Comment (ShimiSun) [reply src/js/components/DataTable/stories/ColumnSize.js:66]: "standard sizes" sounds like a better fit for me. thanks!

File: src/js/components/DataTable/Header.js [hunk 7]
```diff
@@ -130,7 +130,7 @@ const Header = ({
                 pad={pad}
                 plain
                 scope="col"
-                size={size} // Initial size if specified
+                size={widths && widths[property] ? undefined : size}
```
Comment (ShimiSun) [inline src/js/components/DataTable/Header.js:133]: @ericsoderberghp I need your review here.

Comment (ericsoderberghp) [reply src/js/components/DataTable/Header.js:133]: looks right to me

File: src/js/components/DataTable/stories/ColumnSize.js (hunk 3 above)
Comment (ShimiSun) [reply src/js/components/DataTable/stories/ColumnSize.js]: GReat, thanks!

**Comment by ShimiSun** [comment]
Thank you for your contribution @raj-chelur !  Great work!
